LLama/LLamaExecutorBase.cs

```diff
@@ -250,6 +250,14 @@ protected virtual void TryReuseMathingPrefix()
         /// <returns></returns>
         public abstract ExecutorBaseState GetStateData();
 
+        
+        /// <summary>
+        /// Resets the executor to its initial state.
+        /// Note: Does not affect the context and KV cache.
+        /// </summary>
+        /// <returns></returns>
+        public abstract void ResetState();
```

Comment:
What's the purpose of resetting an executor to the initial state, rather than just constructing a new one?